src/containers/downloads/sagas/downloadCases.js

```diff
@@ -0,0 +1,235 @@
+        const dateCaseCreated = formatAsDate(dateTimeCaseCreated);
+        const notes = getPropertyValue(crcCase, [NOTES, 0]);
+
+        const crcCaseEKID :?UUID = getEntityKeyId(crcCase);
+        const caseNeighbors = crcCaseNeighbors.get(crcCaseEKID, List());
+
+        const personNeighbor :Map = caseNeighbors.find((neighbor :Map) => getNeighborESID(neighbor) === peopleESID);
+        const personName :string = getPersonName(getNeighborDetails(personNeighbor));
+
+        const staffList :List = staffByCRCEKID.get(crcCaseEKID, List());
+        const staffPersonName :string = getPersonName(staffList.get(0));
+
+        const statusList :List = statusesByCRCEKID.get(crcCaseEKID, List());
+        const intake = statusList.find((status :Map) => getPropertyValue(status, [PropertyTypes.STATUS, 0]) === INTAKE);
+        let intakeDate = '';
+        if (isDefined(intake)) {
+          const intakeDateTime = getPropertyValue(intake, [EFFECTIVE_DATE, 0]);
+          intakeDate = formatAsDate(intakeDateTime);
+        }
+        const circle = statusList.find((status :Map) => getPropertyValue(status, [PropertyTypes.STATUS, 0]) === CIRCLE);
+        let circleDate = '';
+        if (isDefined(circle)) {
+          const circleDateTime = getPropertyValue(circle, [EFFECTIVE_DATE, 0]);
+          circleDate = formatAsDate(circleDateTime);
+        }
+
+        const forms :List = caseNeighbors
+          .filter((neighbor :Map) => getNeighborESID(neighbor) === formESID)
+          .map((neighbor :Map) => getNeighborDetails(neighbor));
+        const repairHarmAgreement :?Map = forms
+          .find((form :Map) => getPropertyValue(form, [NAME, 0]) === REPAIR_HARM_AGREEMENT);
+        let rhExpirationDate = '';
+        if (isDefined(repairHarmAgreement)) {
+          const expirationDateTime = getPropertyValue(repairHarmAgreement, [DUE_DATE, 0]);
+          rhExpirationDate = formatAsDate(expirationDateTime);
+        }
+
+        const tableRow = OrderedMap()
+          .set(HEADERS.personName, personName)
+          .set(HEADERS.dateAssigned, dateCaseCreated)
+          .set(HEADERS.intake, intakeDate)
+          .set(HEADERS.circle, circleDate)
+          .set(HEADERS.rhExpirationDate, rhExpirationDate)
+          .set(HEADERS.notes, notes)
+          .set(HEADERS.caseManager, staffPersonName);
+        mutator.push(tableRow);
+      });
+    })
+      .sortBy((row) => row.get(HEADERS.personName, '').split(' ')[1]);
+
+    const csv = Papa.unparse(dataTable.toJS());
+    const blob = new Blob([csv], {
+      type: 'application/json'
+    });
+
+    let fileName = '';
+    if ((openCasesIncluded && closedCasesIncluded)
+        || (!openCasesIncluded && !closedCasesIncluded)) fileName = 'CRC_All_Cases';
+    else if (!openCasesIncluded) fileName = 'CRC_Closed_Cases';
+    else if (!closedCasesIncluded) fileName = 'CRC_Open_Cases';
+    if (selectedStaffMember.length) fileName = `${fileName}_${selectedStaffMember.split(' ').join('_')}`;
```

Comment:
hmm that's a good point, but i actually feel like it's not that necessary to move it out to a separate function, since that function won't be reused across components, since each download has different data/filename structure